fastf1/core.py

```diff
@@ -2548,65 +2548,122 @@ def get_weather_data(self) -> pd.DataFrame:
             return pd.DataFrame(columns=self.session.weather_data.columns)
 
     def pick_lap(self, lap_number: int) -> "Laps":
-        """Return all laps of a specific LapNumber in self based on LapNumber
+        """Return all laps of a specific LapNumber in self based on LapNumber.
 
-            lap_1 = ff1.pick_lap(1)
-            lap_25 = ff1.pick_lap(25)
+        .. deprecated:: 3.0.4
+            pick_lap is deprecated and will be removed in a
+            future release. Use :func:`pick_laps` instead.
+
+            lap_1 = session_laps.pick_lap(1)
+            lap_25 = session_laps.pick_lap(25)
 
         Args:
             lap_number (int): Lap number
 
         Returns:
             instance of :class:`Laps`
         """
+        warnings.warn(("pick_lap is deprecated and will be removed in a "
+                       "future release. Use pick_laps instead."),
+                      DeprecationWarning)
         return self[self['LapNumber'] == lap_number]
 
+    def pick_laps(self, lap_numbers: Union[int, Iterable[int]]) -> "Laps":
+        """Return all laps of a specific LapNumber or a list of LapNumbers
+        in self. ::
+
+            lap_1 = session_laps.pick_laps(1)
+            lap_10_to_20 = session_laps.pick_lap(range(10, 21))
+
+        Args:
+            lap_number: int for matching a single lap,
+            a iterable of ints for matching multiple laps
+
+        Returns:
+            instance of :class:`Laps`
+        """
+        if isinstance(lap_numbers, int):
```

Comment:
You could shorten this considerably by doing

```
if isinstance(lap_numbers, (int, float)):
    lap_numbers = [lap_numbers, ]
```

And then just keep the logic for lists of values. 

Advantages: less code, easier readability, no early returns, less duplication and better test coverage
Disadvantage: likely a minimal performance decrease that doesn't matter in code like this

The logic here is not very complicated, so it's not actually that critical. But in general, this is a nice pattern for cases like this. You'll frequently find this in matplotlib and pandas source code.

You could do the same for ``pick_drivers``. For ``pick_teams`` it doesn't really make sense, because it won't reduce complexity.

I hope I'm not being too nitpicky here with small things like this. 😅

Reply:
No worries at all. The maintainer should be nitpicky about their code base haha. I was coding straight to my mental spec there and missed this simplification. Thanks for pointing it out.

Reply:
Ok, good. I felt a bit bad criticizing this and pushing you through another round of review.